main.tf

```
@@ -1169,3 +1169,30 @@ module "dns_hostname" {
 
   context = module.this.context
 }
+
+data "aws_lb_listener" "http" {
+  load_balancer_arn = var.loadbalancer_is_shared ? var.shared_loadbalancer_arn : aws_elastic_beanstalk_environment.default.0.load_balancers[0]
```

```suggestion
  load_balancer_arn = var.loadbalancer_is_shared ? var.shared_loadbalancer_arn : one(module.elastic_beanstalk_environment.default[*].load_balancers)
```

Pretty sure this works but haven't tested it.